enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/StoreResource.java

```diff
@@ -51,19 +52,19 @@
 
     ReadableByteChannel open() throws IOException
     {
-        Optional<PagedFile> existingMapping = pageCache.getExistingMapping( file );
-        if ( existingMapping.isPresent() )
+        boolean isDefaultFileSystem = pageCache.getCachedFileSystem() instanceof DefaultFileSystemAbstraction;
+        if ( !isDefaultFileSystem )
```

Comment:
please just reverse this condition (switch if and else blocks to avoid negation)

Reply:
This is sadly the only way I found to get rid of the otherwise duplicate code. Current contract with the `pageCache.getExistingMapping()` is that if `isPresent()` is true, you have to call `close()` on the object inside the optional.